src/android/IabHelper.java

```diff
@@ -491,10 +491,14 @@ public void onProductDetailsResponse(BillingResult billingResult, List<ProductDe
                 for(ProductDetails productDetails: productDetailsList) {
                     if(sku.equals(productDetails.getProductId())) {
                         int selectedOfferIndex = 0; // Always the first one for us
-                        String offerToken = productDetails
-                           .getSubscriptionOfferDetails()
-                           .get(selectedOfferIndex)
-                           .getOfferToken();
+                        String offerToken="";
+                        List<ProductDetails.SubscriptionOfferDetails> subscriptionDetails = productDetails
+                            .getSubscriptionOfferDetails();
+                        if (subscriptionDetails!=null){
+                            offerToken = subscriptionDetails
+                              .get(selectedOfferIndex)
+                              .getOfferToken();
+                         }
```

Comment:
When `subscriptionDetails` is `null`, `.setOfferToken` will be passed an empty string below. What is the effect of that?

Reply:
it wont effect at all as previously if getSubscriptionOfferDetails was null then it was generating  an exception on .get(selectedOfferIndex) and was not returning any response but i tested if offerToken is blank then it does not generate any error and work as expected so i left it like that but let me if you want me to change it

Reply:
I'm not the maintainer, so my opinion isn't that important. But setting the offer token to an empty string feels wrong. Who knows what other code is dealing with that offer token later (perhaps in cases that you haven't tested) and gets confused by the empty string?

Reply:
offer token is not being used any where else in this method

Reply:
Not in this method, no. But it is passed to Google SDK code with `.setOfferToken` and then who knows how the Google code uses it.

Reply:
on latest commit changed it to better form let me know if its oky

Reply:
It's better. Thanks! Of course, I'm not the maintainer. @wccrawford, the owner of this repo, appears to make changes occasionally. Let's hope he takes into account your suggestion.